pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriter.java

```diff
@@ -69,6 +70,10 @@
                     new Exception("Transaction log buffered write has closed")
             );
 
+    private static final AtomicReferenceFieldUpdater<TxnLogBufferedWriter, TxnLogBufferedWriter.State> STATE_UPDATER =
```

Comment:
Something doesn't add up here. In order to prevent race conditions upon updates to `state` STATE_UPDATER was created. Why then some updates are done without it?